### PR TITLE
refactor(eval-author): simplify dataset augmentation

### PR DESCRIPTION
**Pull request by aleckhoury**
## Summary

Refactors Eval Author’s portable metric handoff and moves Experimentalist to metric-contract-driven optimization. Regression metrics are selector context rather than a hard eligibility gate, while candidates must still report every Pareto objective.

## Changes

- Stage and hand off authored Eval Author datasets and metric contracts.
- Use objective-aware Pareto ranking, direction-aware peer analysis, and LLM-guided soft regression constraints.
- Safely replace Harbor task directories only after validating the staged copy.
- Update metric configuration documentation and focused regression coverage.

## Type of Change

- [x] Code change with documentation updates
- [ ] Documentation only
- [ ] Contributor tooling or automation
- [ ] CI, build, or test infrastructure

## Quality Gates

- [x] Tests added or updated for changed behavior
- [x] Documentation updated for user-visible behavior

## Verification

- [x] Pull request title follows the repository’s Conventional Commit format
- [x] Every commit includes an appropriate `Signed-off-by:` trailer
- [x] Targeted tests pass
- [x] No secrets, API keys, or credentials are included
- [ ] `uv run pre-commit run -a` passes — blocked by 38 existing type-check diagnostics from rebased `main` files outside this PR.

Targeted validation:

- `.venv/bin/pytest plugins/nemo-experimentalist/tests/test_metric_contract.py plugins/nemo-experimentalist/tests/experimentalist/test_dataset_staging.py plugins/nemo-experimentalist/tests/test_experimentalist_analyzer.py plugins/nemo-experimentalist/tests/experimentalist/test_terminator.py -v` — 32 passed.
- `.venv/bin/pytest plugins/nemo-experimentalist/tests/test_experimentalist_backend.py -v` — 19 passed.
- Focused Ruff lint and formatting checks — passed.
- `uv run generate-config-docs && git diff --exit-code docs/set-up/config-reference.mdx` — passed.

### CODE REVIEW

File: plugins/nemo-eval-author/src/nemo_eval_author_plugin/eval_author/agent.py [hunk 1]
```diff
@@ -391,18 +402,42 @@ async def _run(
...
             try:
-                await materialized_dataset.validate()
+                validation_errors: list[str] = []
+                for label, dataset in (
+                    ("train", train_dataset),
+                    ("validation", validation_dataset),
+                    ("insight", materialized_dataset),
+                ):
+                    try:
+                        await dataset.validate()
```
Comment (aleckhoury) [inline plugins/nemo-eval-author/src/nemo_eval_author_plugin/eval_author/agent.py:428]: Is this from harbor?

File: plugins/nemo-eval-author/src/nemo_eval_author_plugin/eval_author/models.py [hunk 2]
```diff
@@ -1,10 +1,51 @@
...
+from typing import Self
 
 from nemo_experimentalist_plugin.entities import Dataset
-from pydantic import BaseModel, ConfigDict, Field
+from pydantic import BaseModel, ConfigDict, Field, field_validator, model_validator
+
+_GENERIC_METRIC_KEYS = frozenset({"reward", "score"})
+
+
+def _validate_metric_keys(value: tuple[str, ...]) -> tuple[str, ...]:
```
Comment (aleckhoury) [inline plugins/nemo-eval-author/src/nemo_eval_author_plugin/eval_author/models.py:14]: Do we have a validation lib for this perhaps? Same for non-empty

**Review by callingmedic911 — APPROVED** [review]
